src/Octoshift/Services/AwsApi.cs

```diff
@@ -45,7 +54,15 @@ private static AmazonS3Client BuildAmazonS3Client(string awsAccessKeyId, string
 
     public virtual async Task<string> UploadToBucket(string bucketName, string fileName, string keyName)
     {
-        await _transferUtility.UploadAsync(fileName, bucketName, keyName);
+        try
+        {
+            await _transferUtility.UploadAsync(fileName, bucketName, keyName);
+        }
+        catch (Exception ex) when (ex is TaskCanceledException or TimeoutException)
+        {
+            throw new OctoshiftCliException($"Upload of archive \"{fileName}\" to AWS timed out", ex);
```

Comment:
I like that we catch this and return a specific clear error message ❤️ 

Could we add the timeout into this error message so it's clear exactly what happened, e.g. "Upload...to AWS timed out after 5 minutes"?

Reply:
Not sure about the "after 5 minutes" wording. I think it could timeout for reasons unrelated to the 5 min timeout set in this PR, like user network connection.